### PR TITLE
applications: asset_tracker: Remove the Asset Tracker application

### PR DESCRIPTION
**Pull request by jhn-nordic**
Removing the deprecated Asset Tracker application.

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

**Comment by jhn-nordic** [comment]
Done in agreement with product management.

**Review by coderbyheart — APPROVED** [review]

File: samples/nrf9160/lte_ble_gateway/README.rst [hunk 1]
```diff
@@ -44,8 +44,6 @@ Two buttons and two switches are used to enter a pairing pattern to associate a
 
 When the connection is established, set switch 2 to **N.C.** to send simulated GNSS data to nRF Cloud once every 2 seconds.
 
-See the :ref:`asset_tracker_user_interface` in the :ref:`asset_tracker` documentation for detailed information about the different LED states used by the sample.
```
Comment (jtguggedal) [inline samples/nrf9160/lte_ble_gateway/README.rst:47]: @nordic-hani FYI, the gateway sample docs should probably be updated when this reference is removed

**Review by jtguggedal — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

**Review by jorgenmk — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
It seems that test repo needs an update as well?

**Comment by jhn-nordic** [comment]
> It seems that test repo needs an update as well?
Yes, testers are in the loop

**Review by plskeggs — APPROVED** [review]
I'll approve but please respond to comments, thanks.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -74,6 +74,11 @@ nRF9160: Asset Tracker v2
 * Added support for A-GPS filtered ephemerides.
 * Added new documentation for :ref:`asset_tracker_v2_util_module` and :ref:`api_modules_common`.
 
+nRF9160: Asset Tracker
+----------------------
+
+* Application is removed
```
[the remark this answers is not in the document]
Comment (OeySa) [reply doc/nrf/releases/release-notes-changelog.rst:80]: Had a quick look through Infocenter and it looks like this is handled already.

**Review by OeySa — APPROVED** [review]

**Comment by jorgenmk** [comment]
@jhn-nordic Oncommit CI is green

**Comment by jhn-nordic** [comment]
doc commit squashed into main commit.

File: doc/nrf/ug_nrf_cloud.rst [hunk 3]
```diff
@@ -41,7 +41,6 @@ Applications and samples
 
 The following applications use the :ref:`lib_nrf_cloud` for services in |NCS|:
```
[the remark this answers is not in the document]
Comment (divipillai) [reply doc/nrf/ug_nrf_cloud.rst]: fixed.

**Review by divipillai — APPROVED** [review]

**Pull request merged**